GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,89 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep
+    {
+        public const string LooseObjectsFileName = "loose-objects.time";
+        private readonly bool forceRun;
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool forceRun = false)
+            : base(context, gitObjects: null, requireObjectCacheLock: requireCacheLock)
+        {
+            this.TimeFilePath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsFileName);

```

Comment:
Since you are using the `virtual` pattern instead of the `readonly` pattern, you could put this down after `TimeBetweenRuns` as

```
protected override string TimeFilePath => Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsFileName);
```